lib/internal/stream_base_commons.js

```diff
@@ -77,15 +77,16 @@ function onWriteComplete(status) {
 
   const stream = this.handle[owner_symbol];
 
-  if (stream.destroyed) {
+  if (status < 0) {
+    const ex = errnoException(status, 'write', this.error);
     if (typeof this.callback === 'function')
-      this.callback(null);
+      this.callback(ex);
     return;
   }
 
-  if (status < 0) {
-    const ex = errnoException(status, 'write', this.error);
-    stream.destroy(ex, this.callback);
+  if (stream.destroyed) {
```

Comment:
We don't want to swallow errors so the `stream.destroyed` case is moved *after* the error case.

Reply:
@mcollibna: I'd like to add a test to this but I'm not really sure how/where?

Reply:
I removed this change for now. Separate PR.